game/scripts/vscripts/events.lua

```diff
@@ -35,6 +35,19 @@ function GameMode:OnGameRulesStateChange(keys)
   DebugPrintTable(keys)
 
   local newState = GameRules:State_Get()
+  -- Pre-Game started
+  if newState == DOTA_GAMERULES_STATE_PRE_GAME then
+    print("Pre-game started")
+    -- initialize modules
+    InitModule(PointsManager)
+    InitModule(Gold)
+    InitModule(BlinkBlock)
+    InitModule(ZoneControl)
+    InitModule(AbilityLevels)
+    InitModule(HeroProgression)
+    InitModule(SellBlackList)
+    InitModule(NGP)
+  end
```

Comment:
You mean put a new `OnPreGame` function in `gamemode.lua` and call it from `events.lua`?